mono/utils/mono-log-windows.c

```diff
@@ -0,0 +1,129 @@
+/*
+ * mono-log-windows.c: Simplistic simulation of a syslog logger for Windows
+ *
+ * This module contains the Windows syslog logger interface
+ *
+ * Author:
+ *    Neale Ferguson <neale@sinenomine.net>
+ *
+ */
+#include <config.h>
+
+#ifdef HAVE_UNISTD_H
+#include <unistd.h>
+#endif
+
+#ifdef HOST_WIN32
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <ctype.h>
+#include <string.h>
+#include <glib.h>
+#include <errno.h>
+#include <time.h>
+#include "mono-logger.h"
+
+static FILE *logFile = NULL;
+static void *logUserData = NULL;
+static char *logFileName = L".//mono.log";
+
+/**
+ * mapSyslogLevel:
+ * 	
+ * 	@level - GLogLevelFlags value
+ * 	@returns The equivalent character identifier
+ */
+static inline char 
+mapLogFileLevel(GLogLevelFlags level) 
+{
+	if (level & G_LOG_LEVEL_ERROR)
+		return ('E');
+	if (level & G_LOG_LEVEL_CRITICAL)
+		return ('C');
+	if (level & G_LOG_LEVEL_WARNING)
+		return ('W');
+	if (level & G_LOG_LEVEL_MESSAGE)
+		return ('N');
+	if (level & G_LOG_LEVEL_INFO)
+		return ('I');
+	if (level & G_LOG_LEVEL_DEBUG)
+		return ('D');
+	return ('I');
+}
+
+/**
+ * mono_log_open_logfile
+ * 	
+ *	Open the logfile. If the path is not specified default to stdout. If the
+ *	open fails issue a warning and use stdout as the log file destination.
+ *
+ * 	@ident - Identifier: ignored
+ * 	@userData - Not used
+ */
+void
+mono_log_open_syslog(const char *ident, void *userData)
+{
+	logFile = fopen(logFileName, "w");
+	if (logFile == NULL) {
+		g_warning("opening of log file %s failed with %s",
+			  strerror(errno));
+	}
+	logUserData = userData;
+}
+
+/**
+ * mono_log_write_logfile
+ * 	
+ * 	Write data to the log file.
+ *
+ * 	@domain - Identifier string
+ * 	@level - Logging level flags
+ * 	@format - Printf format string
+ * 	@vargs - Variable argument list
+ */
```

Comment:
Fixed

From: Johan Lorensson +ADw-notifications+AEA-github.com+ADw-mailto:notifications+AEA-github.com+AD4APg-
Reply-To: mono/mono +ADw-reply++0004b6fa8e817be993ed9d1f98484b9566a6483031e3003e92cf000000011369030192a170ce03ea18AQA-reply.github.com+ADw-mailto:reply++0004b6fa8e817be993ed9d1f98484b9566a6483031e3003e92cf000000011369030192a170ce03ea18AQA-reply.github.com+AD4APg-
Date: Friday, June 3, 2016 at 4:34 AM
To: mono/mono +ADw-mono+AEA-noreply.github.com+ADw-mailto:mono+AEA-noreply.github.com+AD4APg-
Cc: Neale Ferguson +ADw-neale+AEA-sinenomine.net+ADw-mailto:neale+AEA-sinenomine.net+AD4APg-, Mention +ADw-mention+AEA-noreply.github.com+ADw-mailto:mention+AEA-noreply.github.com+AD4APg-
Subject: Re: +AFs-mono/mono+AF0- Enhance log tracing (+ACM-2734)


In mono/utils/mono-log-windows.c+ADw-https://github.com/mono/mono/pull/2734+ACM-discussion+AF8-r65673318+AD4-:

+AD4-               g+AF8-warning(+ACI-opening of log file +ACU-s failed with +ACU-s+ACI-,
+AD4-                         strerror(errno))+ADs-
+AD4-       +AH0-
+AD4-       logUserData +AD0- userData+ADs-
+AD4- +AH0-
+AD4-
+AD4- +ACoAKg-
+AD4-  +ACo- mono+AF8-log+AF8-write+AF8-logfile
+AD4-  +ACo-
+AD4-  +ACo-    Write data to the log file.
+AD4-  +ACo-
+AD4-  +ACo-    +AEA-domain - Identifier string
+AD4-  +ACo-    +AEA-level - Logging level flags
+AD4-  +ACo-    +AEA-format - Printf format string
+AD4-  +ACo-    +AEA-vargs - Variable argument list
+AD4-  +ACo-/


Comment doesn't match function, copy paste?

+IBQ-
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub+ADw-https://github.com/mono/mono/pull/2734/files/583d1a2580a5fc61d0002920c17c23fa71375c3d+ACM-r65673318+AD4-, or mute the thread+ADw-https://github.com/notifications/unsubscribe/AAS2-vS64huue9v-6HLzvqVlGRhsml3Uks5qH-cBgaJpZM4HsGtY+AD4-.